drivers/lora/sx12xx_common.c

```diff
@@ -249,9 +294,36 @@ int sx12xx_lora_send_async(const struct device *dev, uint8_t *data,
 	/* Store signal */
 	dev_data.operation_done = async;
 
-	Radio.SetMaxPayloadLength(MODEM_LORA, data_len);
+	if (dev_data.tx_cfg.cad) {
+
+		uint8_t *buffer = k_malloc(data_len);
```

Comment:
No `k_malloc/k_free` in the driver please, since that adds a dependency on the system heap that does not currently exist.
You are already happy to add `k_sleep(K_MSEC(delay));` in the `lora_send_async` function, why is waiting for CAD to finish a deal breaker?